File: JavaScript/222. Count Complete Tree Nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var countNodes = function(root) {
    var n1 = 0;
    var n2 = 0;
    var l = root;
    var r = root;
    while(l!=null){
        n1++;
        l=l.left;
    }
    while(r!=null){
        n2++;
        r=r.right;
    }
    if(n1==n2){
        return Math.pow(2,n1)-1;
    }
    return 1+countNodes(root.left)+countNodes(root.right);
};

File: C#/222. Count Complete Tree Nodes.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int CountNodes(TreeNode root) {
        int n1 = 0;
        int n2 = 0;
        TreeNode l = root;
        TreeNode r = root;
        while(l!=null){
            n1++;
            l=l.left;
        }
        while(r!=null){
            n2++;
            r=r.right;
        }
        if(n1==n2){
            return (int)Math.Pow(2,n1)-1;
        }
        return 1+CountNodes(root.left)+CountNodes(root.right);
    }
}
